gems/pending/appliance_console/database_replication.rb

```diff
@@ -0,0 +1,128 @@
+require 'appliance_console/logging'
+require 'appliance_console/prompts'
+require 'pg'
+require 'English'
+
+module ApplianceConsole
+  class DatabaseReplication
+    include ApplianceConsole::Logging
+
+    REPMGR_CONFIG = '/etc/repmgr.conf'.freeze
+
+    attr_accessor :cluster_name, :node_number,
+                  :database_name, :database_user, :database_password,
+                  :primary_host, :standby_host
+
+    def ask_for_unique_cluster_node_number
+      self.node_number = ask_for_integer("number uniquely identifying this node in the replication cluster")
+    end
+
+    def ask_for_database_credentials
+      ask_for_cluster_database_credentials
+      self.primary_host = ask_for_ip_or_hostname("primary database hostname or IP address", primary_host)
+    end
+
+    def confirm(including_standby_host = nil)
+      clear_screen
+      say(<<-EOL)
+Replication Server Configuration
+
+        Cluster Node Number:        #{node_number}
+        Cluster Database Name:      #{database_name}
+        Cluster Database User:      #{database_user}
+        Cluster Database Password:  "********"
+        Cluster Primary Host:       #{primary_host}
+        EOL
+      say("        Standby Host:               #{standby_host}") unless including_standby_host.nil?
+
+      agree("Apply this Replication Server Configuration? (Y/N): ")
+    end
+
+    def ask_continue_if_already_configured?
+      if File.exist?(REPMGR_CONFIG)
+        say("Warning: File #{REPMGR_CONFIG} exists. Replication is already configured")
+        Logging.logger.info("Warning: File #{REPMGR_CONFIG} exists. Replication is already configured")
+        return agree("Continue with configuration? (Y/N): ")
+      end
+      true
+    end
+
+    def create_config_file(host)
+      File.open(DatabaseReplication::REPMGR_CONFIG, "w") do |f|
+        f.puts("cluster=#{cluster_name}")
+        f.puts("node=#{node_number}")
+        f.puts("node_name=#{host}")
+        f.puts("conninfo='host=#{host} user=#{database_user} dbname=#{database_name}'")
+        f.puts("use_replication_slots=1")
+        f.puts("pg_basebackup_options='--xlog-method=stream'")
+      end
+      true
+    end
+
+    def generate_cluster_name
+      begin
+        pg_conn = PG::Connection.new(:dbname   => database_name,
+                                     :host     => primary_host,
+                                     :user     => database_user,
+                                     :password => database_password)
+        primary_region_number =
+          pg_conn.exec("SELECT last_value FROM miq_databases_id_seq").first["last_value"].to_i / 1_000_000_000_000
+        self.cluster_name = "miq_region_#{primary_region_number}_cluster"
+      rescue PG::ConnectionBad, PG::UndefinedTable, PG::SyntaxError => e
```

Comment:
If the user enters the wrong password by mistake not rescuing PG::ConnectionBad causes appliance_console to  fail with a traceback. The way it is now the user is presented with a message that they password was wrong and they can start over.